tmt/steps/provision/connect.py

```diff
@@ -110,8 +111,14 @@ def go(self):
             data['key'] = key
 
         # And finally create the guest
-        self._guest = tmt.Guest(data, name=self.name, parent=self.step)
+        self._guest = GuestConnect(data, name=self.name, parent=self.step)
 
     def guest(self):
         """ Return the provisioned guest """
         return self._guest
+
+
+class GuestConnect(tmt.Guest):
```

Comment:
Hm, is this new class necessary? As far as I know, the `Guest` class was used for `provision -h connect`.

Reply:
I think: it doesn't matter but it looks much better when you have prepared class which you can edit in future. And it doesn't looks too much generic.

Reply:
The extra class does not make sense to me either. It does not bring anything new. +1 for removing it.